plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/LambdaExp.kt

```diff
@@ -27,15 +25,21 @@ class LambdaExp(
 
     override fun insertCallImpl(args: List<ExpEmbedding>, ctx: StmtConversionContext<ResultTrackingContext>): ExpEmbedding =
         ctx.withResult(returnType) {
+            val inlineBody = function.body ?: throw Exception("Lambda $function has a null body")
             val paramNames = function.valueParameters.map { it.name }
-            paramNames.forEach { ctx.addScopedName(it) }
-            val callArgs = ctx.getFunctionCallSubstitutionItems(args)
+            val callArgs = ctx.getInlineFunctionCallArgs(args)
             val subs = paramNames.zip(callArgs).toMap()
-            val lambdaCtx = this.newBlock().withLambdaContext(this.signature, this.resultCtx.resultVar.name, subs, scopedNames)
-            lambdaCtx.convert(function.body!!)
-            // NOTE: It is necessary to drop the last stmt because is a wrong goto
-            val sqn = lambdaCtx.block.copy(stmts = lambdaCtx.block.stmts.dropLast(1))
+            val returnLabelName = ReturnLabelName(newWhileIndex())
```

Comment:
why are you using a `whileIndex` for the return label?

Reply:
We need some value to use for the return label name, and I don't have as easy access to the scope index here as I did previously.  I have a follow-up PR where I do this more properly.